### ISSUE TITLE
Error 477

### ISSUE THREAD

**Issue by leon1995** [nborrmann/jodel_api#23]
Hi,
I´ve downloaded the newest version and still got the Error 477 while creating an account.
I added the test result. maybe it will help you to solve my problem :)

Thx!
Leon
[testresult.txt](https://github.com/nborrmann/jodel_api/files/992349/testresult.txt)

**Comment by nborrmann** [comment]
Please post the output of `pip freeze`.

**Comment by leon1995** [comment]
Thx for your incredible fast reply!
Here pip freeze output:

adium-theme-ubuntu==0.3.4
argcomplete==0.8.1
argh==0.26.1
backports.ssl-match-hostname==3.4.0.2
beautifulsoup4==4.4.1
blinker==1.3
certifi==2015.11.20.1
chardet==2.3.0
click==6.2
colorama==0.3.7
ConfigArgParse==0.10.0
construct==2.5.2
cryptography==1.2.3
enum34==1.1.2
feedparser==5.1.3
funcsigs==1.0.2
future==0.16.0
hpack==2.1.0
html2text==2016.1.8
html5lib==0.999
idna==2.0
ipaddress==1.0.16
jodel-api==1.1.3
lxml==3.5.0
mitmproxy==0.15
mock==2.0.0
netlib==0.15.1
passlib==1.6.5
pathtools==0.1.2
pbr==2.1.0
Pillow==3.1.2
pyasn1==0.1.9
pycurl==7.43.0
pygpgme==0.3
PyMySQL==0.7.2
pyOpenSSL==0.15.1
pyparsing==2.0.3
pyperclip==1.5.11
PyYAML==3.11
requests==2.13.0
scour==0.32
six==1.10.0
tornado==4.2.1
unity-lens-photos==1.0
urwid==1.3.1
uTidylib==0.2
watchdog==0.8.3

**Comment by nborrmann** [comment]
Please also give me the output of 

```
$ pip -V
$ python -V
$ python -c "import jodel_api; print(jodel_api.JodelAccount.secret)"
```

**Comment by leon1995** [comment]
`1 pip 8.1.1 from /usr/lib/python2.7/dist-packages (python2.7)`
`Python 2.7.12`
`XjwIzRfxCXsVLRkutOHyiexqDnDutUCnTrECnvpw`

**Comment by nborrmann** [comment]
That is weird. The correct secret is `XjwOzRfxCXsVLRkutOHyiexqDnDutUCnTrECnvpw` (`O` instead of `I` in the 4th column), the rest is the same.

I just checked, both github and pypi have the correct secret. Did you install jodel_api manually using `git clone`? Any chance you accidentally flipped that letter?

**Comment by leon1995** [comment]
I checked it twice. It wasnt a typo.
I installed it with github.

![output](https://cloud.githubusercontent.com/assets/15685899/25956138/391eb54a-366b-11e7-8ddd-15fd9949b838.PNG)

i will install it again with github and try again

**Comment by nborrmann** [comment]
The only thing I can still imagine is that your script somehow links against a different version of `jodel_api` than the one installed by pip. Maybe you have an old `jodel_api.py` in the same directory as your script. Maybe try adding the statement `print(jodel_api.JodelAccount.secret)` into your actual script.

**Comment by leon1995** [comment]
I´ve deleted and installed it new. Used `pip uninstall jodel_api`. Also did a new `git clone`. Didnt helped. Then 
`sudo apt-get remove python-pip` 
`sudo apt-get install python-pip`
`pip install jodel_api`
worked. I could successfully create a account (did not try more).

I also tried with the bash on windows 10 and got this error
![winbasherror](https://cloud.githubusercontent.com/assets/15685899/25959661/a9fbe234-3674-11e7-9ebe-01201d17c686.PNG)

I´m installing Xubuntu on a different VM and I´ll try it there. I´ll post the result when finished

**Comment by nborrmann** [comment]
Yeah, looks like it was some version/environment mess up.

**Comment by leon1995** [comment]
When I try to create a post with `j.create_post(message='Hello World', imgpath=None, b64img=None, color=None, ancestor=None, channel="")` after verifying and stuff, I get this
`Expectation: One of message or imgpath must be null`
Ofcause it has to be a none-empty post, but I´ve set a message, or not?

**Comment by nborrmann** [comment]
Yeah that should work. Can't reproduce this issue either. Maybe a typo? The other issue is resolved now I guess? Then I will close the issue for now.

Btw, you don't need to add the parameters that are `None`. Just `j.create_post("Hello World")` will do.

**Comment by gaurav-stepin** [comment]
Hello,
I am having same issue since yesterday. The setup was working fine and then it stopped without any change made on server or in package. I tried all steps mentioned above in the trail but nothing worked. Should I provide output of pip -V etc?

P.S :- I downloaded fresh latest package and setup, but same issue.

Many Thanks,
Gaurav

**Comment by nborrmann** [comment]
Please post the output of the following commands:

```
$ pip show jodel_api
$ pip -V
$ python -V
$ python -c "import jodel_api; print(jodel_api.JodelAccount.version); print(jodel_api.JodelAccount.secret)"
```

**Comment by gaurav-stepin** [comment]
Hi nborrmann,
Thanks for prompt response.
Here are respective commands I fired with the output.

>> pip show jodel_api

Name: jodel-api
Version: 1.1.5
Summary: Unoffical Python Interface to the Jodel API
Home-page: https://github.com/nborrmann/jodel_api
Author: Nils Borrmann
Author-email: n.borrmann@googlemail.com
License: MIT
Location: /usr/local/lib/python2.7/dist-packages/jodel_api-1.1.5-py2.7.egg
Requires: requests, future, mock
---------------------------------------------------------------------------
>> pip -V 

pip 9.0.1 from /home/jaldip/.local/lib/python2.7/site-packages (python 2.7)
---------------------------------------------------------------------------
>> python -V

Python 2.7.12
---------------------------------------------------------------------------
>> python -c "import jodel_api; print(jodel_api.JodelAccount.version); print(jodel_api.JodelAccount.secret)"

4.44.1
DOTrPEbAXvJmqDKnvdLCPJlUyKenbDoAJWiAxxnT

**Comment by nborrmann** [comment]
Fixed in 54e46ef.

**Comment by gaurav-stepin** [comment]
That was a wonderfull help mate.Thanks alot.
